File: rgbimage.m
% Read image and separate color channels
x = imread("/home/rguktrkvalley/Pictures/redcolor.png");
red = x(:,:,1);  % Extract red channel
gr = x(:,:,2);  % Extract green channel
bl = x(:,:,3);  % Extract blue channel

% Save channels to CSV files
csvwrite("red.csv", red);
csvwrite("gr.csv", gr);
csvwrite("bl.csv", bl);

File: rgbimage.py
import cv2
import csv

# Read image
image = cv2.imread("path/to/your/image.png")

# Separate color channels
red, green, blue = cv2.split(image)

# Save channels to CSV files
csv.writerow(open("red.csv", "w"), red.flatten())
csv.writerow(open("gr.csv", "w"), green.flatten())
csv.writerow(open("bl.csv", "w"), blue.flatten())

